server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/service/session/WebSessionManager.java

```diff
@@ -139,19 +153,66 @@ public WebSession getWebSession(@NotNull HttpServletRequest request, @NotNull Ht
         return webSession;
     }
 
-    private void tryToRestorePreviousUserSession(WebSession webSession) {
-        try {
-            SMAuthInfo oldAuthInfo = webSession.getSecurityController().restoreUserSession(webSession.getSessionId());
-            if (oldAuthInfo == null) {
-                return;
+    /**
```

Comment:
Please describe what is the puyrpose of session restoration,

Reply:
In case of replacing the backend node, we lose active valid sessions in the cache. With the next frontend request, we try to restore the lost session in order to continue working with it